R/model.R

```diff
@@ -332,6 +324,23 @@ save_fig.keyATM_viz <- function(x, filename, ...)
 }
 
 
+get_doc_index <- function(docs)
+{
+  lapply(docs, length) %>% unlist(use.names = FALSE) -> len
+  index <- 1:length(docs)
+  nonzero_index <- index[index[len != 0]]
+  zero_index <- index[index[len == 0]]
+  if (length(zero_index) != 0) {
+    warning("Number of documents of 0 length: ", length(zero_index), "\n",
+             "This may cause invalid covariates or time index.", "\n",
```

Comment:
I am not sure if we need this line because the code would drop corresponding covariates and time index.

Reply:
Yeah, I generally do not want users to implicitly drop data. Also I don't know the unexpected effect of dropping covariates (or time stamps).

Reply:
I fully agree that users should be aware that data is dropped. This particular line might cause confusion to the users because the current code actually drops corresponding observations.

Reply:
I see. I'll add a comment. It seems this is automatically merged when you approved the request. I'll send another pull request.

Reply:
Thank you so much! I think something like this works:
`Corresponding covariates/timestamps are automatically dropped but please check them.`

> It seems this is automatically merged when you approved the request.
I realize that it was merged. I have commented but did not approve this...

Reply:
Maybe you need to request a change explicitly.